activerecord/lib/active_record/sanitization.rb

```diff
@@ -144,7 +144,7 @@ def sanitize_sql_like(string, escape_character = "\\")
       #   # => "name='foo''bar' and group_id='4'"
       def sanitize_sql_array(ary)
         statement, *values = ary
-        if values.first.is_a?(Hash) && statement =~ /:\w+/
```

Comment:
This condition was added by [this](https://github.com/rails/rails/commit/554597d6#diff-391caa9b9464021e932ebf657fa9b13cR649). I think the commit did not care about situations where there is arity mismatch between statement and hash.

> do we need, also, to update the doc for the method?

Maybe do not need, because the theme of this commit is how we care error case (arity mismatch).